Nipadhye/firewall packet capture feature

nikhilpadhye1 opened the pull request:
## ARM API Information (Control Plane)
** Closed another PR which was active to move the feature to the correct API Version**
*Link to closed PR: https://github.com/Azure/azure-rest-api-specs/pull/20601*
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [X] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. NRP 170 - 9/23
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. Next release, 9/23 NRP 170 release
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [X] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Commit:
Copy solutions files from 2015-06-01-preview

Commit:
update API version

Commit:
update readme

Commit:
Add required to ExternalSecuritySolution

Commit:
add new swagger files

ariklin committed:
Merge branch 'master' into azure-rest-api-specs-securitySolutions

Commit:
fix errors1

Commit:
Merge branch 'azure-rest-api-specs-securitySolutions' of https://github.com/ariklin/azure-rest-api-specs into azure-rest-api-specs-securitySolutions

Commit:
fix2

Commit:
fix 3

Commit:
fix 4

Commit:
fix4

Commit:
fix4

Commit:
fix conflict

Commit:
fix conflict2

Commit:
fix conflict4

Commit:
fix conflict 6

Commit:
fix conflict 7

Commit:
ttt

Commit:
rrr

Commit:
aaa

Commit:
bbb

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Merge branch 'master' into azure-rest-api-specs-securitySolutions

Commit:
fix 10

Commit:
Prettier check

Commit:
update readme

Commit:
fix reference error

Commit:
fix comment

Commit:
fix conflict

Commit:
fic conflict2

Commit:
fix conflict3

Commit:
fix conflict4

ariklin committed:
Merge branch 'master' into master

ariklin committed:
Merge branch 'master' into azure-rest-api-specs-securitySolutions

ariklin committed:
Merge pull request #2 from ariklin/azure-rest-api-specs-securitySolutions

Azure rest api specs security solutions

Commit:
remove files

Commit:
delete externalSecuritySolutions

Commit:
add assessments.json

ariklin committed:
Merge branch 'master' into master

Commit:
fix readme.md

Commit:
Merge branch 'master' of https://github.com/ariklin/azure-rest-api-specs

Commit:
Fix validation errors 1

Commit:
Fix validation errors 2

Commit:
merge

Commit:
Fix validation errors 3

Commit:
Fix validation errors 4

Commit:
Fix validation errors 5

Commit:
Fix validation errors 5

Commit:
Fix validation errors 5

Commit:
Fix validation errors 6

Commit:
Fix validation errors 4

Commit:
Fix validation errors 7

Commit:
Fix validation errors 8

Commit:
Fix validation errors 9

Commit:
Fix validation errors 10

Commit:
fix validation errors 11

Commit:
fix validation errors 11

Commit:
ix validation errors 12

Commit:
ix validation errors 13

Commit:
fix validation errors 14

ariklin committed:
Merge branch 'master' into master

Commit:
Fix validation errors 15

Commit:
Merge branch 'master' of https://github.com/ariklin/azure-rest-api-specs

Commit:
Fix validation errors 16

Commit:
run prettier

harics24 committed:
Merge pull request #3 from Azure/master

Taking latest from upstream

MehaKaushik committed:
Merge pull request #8 from Azure/master

Merge master

Commit:
adding Services APIs

Commit:
more changes

Commit:
updated examples

amisi01 committed:
Merge pull request #3 from Azure/master

Merging with master

amisi01 committed:
Merge pull request #2 from MehaKaushik/master

Updating Services API

harics24 committed:
Merge branch 'Azure:master' into master

Commit:
Defining the APIs for the new NSP resource

harics24 committed:
Merge branch 'Azure:master' into master

amisi01 committed:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

amisi01 committed:
Merge branch 'Azure:master' into master

amisi01 committed:
Merge branch 'Azure:master' into master

amisi01 committed:
Merge branch 'Azure:master' into master

harics24 committed:
Merge branch 'Azure:master' into master

amisi01 committed:
Merge branch 'Azure:master' into master

amisi01 committed:
Merge branch 'Azure:master' into master

harics24 committed:
Merge branch 'Azure:master' into master

yang-hai-feng committed:
IoT Central Public API 1.1-preview Oct Release (#16514)

* Add new 2.0-preview.1 version for IoT Central

* fix typo on enum value

* rollback 1.0 changes, applied 2.0-preview.1

* fix typo

* remove non-used examples

* fix style

* update version from 2.0-preview.1 to 1.1-preview

* update code review comments, fix patch consumes with json-patch support, fix examples for fileupload

* add nextLink to collection of orgs

* fix typo

* fix wrong position of paging

* fix the paging

* add . to description

* update default to use enum for api-version

* add oct release with query, cde v2, device filter, device template filter, relationship, edge template, simplified template patching

* update examples

* fix lint error, model validation

* fix missing type in examples, remove url format

* fix ID case

* fix id case

* fix ID case

* add iotc query language reference

* fix typo

amisi01 committed:
Merge branch 'Azure:master' into master

amisi01 committed:
Merge branch 'Azure:main' into master

harics24 committed:
Merge branch 'master' into master

ariklin committed:
test

ariklin committed:
SecuritySolution test

ariklin committed:
2

ariklin committed:
org

ariklin committed:
1

ariklin committed:
Merge branch 'master' into master

amisi01 committed:
Merge branch 'master' into master

kesharma committed:
[DataFactory]Add support for running staging on multiple sources in parallel (#18825)

* Add support for running staging on multiple sources in parallel

* update

tshoham committed:
Add default accountid (#19417)

Similar PR was merged into Main.
s360 ticket still leading to this branch - so we are pushing the pull request with the same fixes here.

honghr committed:
Adds base for updating Microsoft.Network from version stable/2022-05-01 to version 2022-07-01

honghr committed:
Updates readme

honghr committed:
Updates API version in new specs and examples

sbhosalemsft committed:
Adds a SaaS NVA swagger changes (#20973)

* add saas nva swagger changes

* add saas nva put example

* add arm-id format

* minor change

* address review comments

* add custom words

nikhilpadhye1 committed:
Moving captures to 2022-07-01 API

nikhilpadhye1 committed:
Adding rest of packet capture api

nikhilpadhye1 commented:
@raosuhas @lirenhe moved the PR to correct API version. Ran into issues trying to rebase previous PR branch so making this new active PR. Addressed all the comments left by @raosuhas in the older PR

nikhilpadhye1 committed:
changing api version in example file to new api version

nikhilpadhye1 committed:
Re-ran prettier check

nikhilpadhye1 committed:
Changing mismatched types

lirenhe commented:
@nikhilpadhye1, could you fix the model validation errors?  The Swagger example has some conflicts with API definition.

raych1 commented:
@nikhilpadhye1 , please fix `modelvalidation` failures.

nikhilpadhye1 committed:
fixing model validation error

msyyc commented:
@nikhilpadhye1 Please merge public main branch into [Azure:release-network-Microsoft.Network-2022-07-01](https://github.com/Azure/azure-rest-api-specs/tree/release-network-Microsoft.Network-2022-07-01) to pass python pipeline

raosuhas commented on specification/network/resource-manager/Microsoft.Network/stable/2022-07-01/azureFirewall.json:
```diff
@@ -437,6 +437,66 @@
+        "responses": {
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          },
```
need to also include the 200 for the final status returned by the location header

raosuhas replied:
Just bring this comment back here from the older PR : https://github.com/Azure/azure-rest-api-specs/pull/20601

This 200 refers to the final state that the client will get when it does the GET on the location header that you return as part of the 202

nikhilpadhye1 replied:
I believe i addressed the changes you asked for
please take a look and request an example if something is incorrect. Thanks!

nikhilpadhye1 replied:
added 200 please let me know if this is what was needed. Or is something else needed?

nikhilpadhye1 committed:
Adding 200 status code

nikhilpadhye1 committed:
Fixed last model validation issue

nikhilpadhye1 commented:
> @nikhilpadhye1 Please merge public main branch into [Azure:release-network-Microsoft.Network-2022-07-01](https://github.com/Azure/azure-rest-api-specs/tree/release-network-Microsoft.Network-2022-07-01) to pass python pipeline

I tried to merge public main into the release branch and got up-to-date message from command-line @msyyc

nikhilpadhye1 committed:
Adding 200 status code

nikhilpadhye1 committed:
Added 200 to wrong api changing accordingly

nikhilpadhye1 reviewed (commented):
Addressed all comments, please review and help close this PR

nikhilpadhye1 commented:
@lirenhe @raosuhas  is there anymore feedback for the PR? Can we close this?

nikhilpadhye1 committed:
Update azureFirewall.json

Typo

nikhilpadhye1 committed:
Update AzureFirewallPacketCapture.json

typo

saisujithreddym commented:
@MikhailTryakhov @raosuhas @lirenhe @raych1 Can we please get the review on the PR?

lirenhe commented:
@nikhilpadhye1, Please work with ARM team to address their feedback.
I also see the PR to merge from network 2022-07-01 to main branch was already created. 
Could you work with @MikhailTryakhov about the timeline for this one?

davsta90 committed:
doc and example fixes (#21558)

* doc and example fixes

* Prettier whitespace changes

* trying to fix Newtonsoft.Json.JsonSerializationException: Error converting value NoExplanation to type 'System.Collections.Generic.Dictionary

* attempting to fix validaiton errors

* attempting to fix validaiton errors

touchdreamer committed:
snowflake swagger (#21552)

Co-authored-by: Zehua Chen <zehuachen@microsoft.com>

omaremadeldin committed:
[Cognitive Services / Language] Dynamic classification `class` to `classifications` (#21550)

* Dynamic classification `class` to `classifications`

* Update analyzetext.json

* Update analyzetext.json

* Update analyzetext.json

* update DynamicClassification x-ms-examples

Co-authored-by: Krista Pratico <krpratic@microsoft.com>

jaimeva-msft committed:
Add npmrc as custom word (#21593)

Wzb123456789 committed:
Update readme.python.md (#21599)

Wzb123456789 committed:
[Resource] Update readme.python.md (#21600)

* Update readme.python.md

* Update readme.python.md

Co-authored-by: Zhenbiao Wei (WICRESOFT NORTH AMERICA LTD) <v-zhenbwei@microsoft.com>

qianwens committed:
[service connector]add api version 2022-11-01-preview (#21408)

* add api version 2022-11-01-preview

* fix api versions in sample

* fix sample

* add dryrun to custom words

* update configurationNames API

Co-authored-by: xiaofanzhou <xiaofanzhou@microsoft.com>

Anandaraj-Selvam committed:
lint error fix (#21594)

Co-authored-by: Anandaraj Selvam <anselvam@microsoft.com>

Alancere committed:
[Go] loadtestservice rename module-name (#21602)

gaoyp830 committed:
Minor fix on billingBeneifts (#21566)

* Minor fix on billingBeneifts

* Update billingbenefits.json

* Update billingbenefits.json

* Update billingbenefits.json

* Update billingbenefits.json

* Update billingbenefits.json

* fix examples

* Update billingbenefits.json

anat-gilenson committed:
Release securityinsights microsoft.security insights 2022 11 01 (#21540)

* Adds base for updating Microsoft.SecurityInsights from version stable/2022-08-01 to version 2022-11-01

* Updates readme

* Updates API version in new specs and examples

* SecurityMLAnalyticsSettings to Microsoft.SecurityInsights-2022-11-01-stable (#20944)

* update

* update api versions

* Automation rules 2022-11-01 - alert & update triggers (#21205)

* Automation rules 2022-11-01 stable version

* preetier

* Remove unused url parameters

Co-authored-by: jungph808 <100548649+jungph808@users.noreply.github.com>
Co-authored-by: Dor Siso <60354892+dosiso@users.noreply.github.com>

dejimarquis committed:
[Hub Generated] Review request for Microsoft.DigitalTwins to add version stable/2022-10-31 (#20374)

* Adds base for updating Microsoft.DigitalTwins from version stable/2022-05-31 to version 2022-10-31

* Updates readme

* Updates API version in new specs and examples

* Introducing 22-10-31 control plane changes for ADt

* Fixing validation errors

* making userAssignedIdentity rx id nullable

* PR feedback

* using common-types

* Revert "using common-types"

This reverts commit 339589c7cbb8221d64af7287255ddb06334930db.

* ManagedIdentity -> ManagedIdentityReference

* Add default table and consumer groups

Co-authored-by: Ayodeji Marquis <aymarqui@microsoft.com>

KalpeshChavan12 committed:
Updated doc (#20626)

Co-authored-by: Kalpesh Chavan <kachavan@microsoft.com>

RakeshMohanMSFT committed:
{containerregistry} Fix #21603: Mention anonymous pull access is a preview feature, available in the Standard and Premium service tiers (#21623)

Fixes https://github.com/Azure/azure-rest-api-specs/issues/21603

shesamian committed:
[Hub Generated] Review request for Microsoft.RecoveryServices to add version stable/2022-10-01 (#20998)

* Adds base for updating Microsoft.RecoveryServices from version stable/2022-09-10 to version 2022-10-01

* Updates readme

* Updates API version in new specs and examples

* Changes introduced in october 1st, 2022

* Adds base for updating Microsoft.RecoveryServices from version stable/2022-09-10 to version 2022-10-01

* Updates readme

* Updates API version in new specs and examples

* Update service.json

* Tiny change that make SDK generation possible

* fix x-ms-identifiers lint error

* fixing lintdiff errors

* fixing lintdiff errors

* complient error response

* error response missing type

* fixed style errors

* better pattern

* Updated the pattern

* Reverted unncessary changes

* Revert parameter name change

* updated examples parameters

krishna1s committed:
Loadtest stableapis (#21612)

* publishing stable api version 2022-11-01 for load test service

* prettier run

* spell check

* changing swagger back to 2.0

* adding description

* prettier changes

* renaming opertation  id

* adding x-ms-enum extension

* validation fixes

* renaming examples

* review changes

* adding x-ms-parameter-location

* review fixes

* removing required field

* adding x-ms-enum values

* prettier fix

* spell fixes

* prettier fix

* adding new metrics and app component apis

* updating examples

* removing list resource type api

* changing to kebab case

* changing x-ms-parameter-location value for sdk

* updating example

* removing resourceId

* removing resource Id from examples

* adding missing desc

* removing read only properties from examples

* spell fix

* review fixes

* mode renaming

* properties changes in examples

* renaming

* Update loadtestservice.json

* renaming properties

* renaming property

* desc changes

* Add readme SDK changes (#21315)

Co-authored-by: Harshan B <harshanb@microsoft.com>

* review changes

1. adding file with application octet stream
2.taking fileName as param instead of fileId
3. adding new api for metrics dimension values
4. simplifying metrics api

* model validation fixes

* spell fixes

* updating desc for search and orderby

* desc changes

* adding 201 response

* adding 201 status code

* sdk consistency changes

* filename renaming

* renaming file info

* adding default pass fail criteria action value

Co-authored-by: Krishna Chandak <krchanda@microsoft.com>
Co-authored-by: Harshan B <Harshan01@gmail.com>
Co-authored-by: Harshan B <harshanb@microsoft.com>

shesamian committed:
[Hub Generated] Review request for Microsoft.RecoveryServices to add version stable/2022-10-01 (#21448)

* Adds base for updating Microsoft.RecoveryServices from version preview/2022-09-30-preview to version 2022-10-01

* Updates readme

* Updates API version in new specs and examples

* Minor markdown linting issue

* Modified the formatting

* used common parameters

sbhat9 committed:
Purview policy store devops policies api specs (#21235)

* Purview devops policies api specs

* fixing the failing checks

* fixing the failing checks

* Fixing the checks

* making few properties read only

* Addressing the review comments

* Addressing the review comments

* Addressing the review comments

* Lint error fix

FumingZhang committed:
Review request for Microsoft.ContainerService to add version 2022-10-02-preview (#21439)

* Adds base for updating Microsoft.ContainerService from version preview/2022-09-02-preview to version 2022-10-02-preview

* Updates readme

* Updates API version in new specs and examples

* Pmiller/nodeosupgradechannel (#21326)

* missed working comment

* matthew comments

* unpatched is a word

* Add MaintenanceWindow into MaintenanceConfigurationProperties (#21337)

* Add MaintenanceWindow into MaintenanceConfigurationProperties

* Fx PrettierCheck

* Fix ARM ID format for samples (#21402)

* Fix Swagger Model Validation for the 2022-10-02-preview dev branch of container service (#21373)

* fix API version

* fix INVALID_FORMAT byte

* fix INVALID_FORMAT arm-id

* fix missing resource id

* fix LRO_RESPONSE_HEADER

* Update readmes for the 2022-10-02-preview dev branch of container service (#21366)

* update readmes

* remove fleets

* Updating abort apis response to make it long running  (#21362)

* Updating response abort abort apis

* Updating abort AP and MC examples

* Changes per CI checks

* Adding x-ms-long-running-operation-options

* Adding location header

Co-authored-by: InduSridhar <indusridhar@microsoft.com>

* fix 202 location header

* fix ProvisioningState validation

* add x-ms-long-running-operation-options

* fix ResourceNameRestriction

* Fix abort location url in example (#21446)

* Updating response abort abort apis

* Updating abort AP and MC examples

* Changes per CI checks

* Adding x-ms-long-running-operation-options

* Adding location header

* Fixing location url in abort examples

Co-authored-by: InduSridhar <indusridhar@microsoft.com>

* Abort api response fix (#21559)

* Updating response abort abort apis

* Updating abort AP and MC examples

* Changes per CI checks

* Adding x-ms-long-running-operation-options

* Adding location header

* Fixing location url in abort examples

* Fixing headers and body for abort response

* Linter fix

* Lint fix

* Updating managedclusters.json responses

Co-authored-by: InduSridhar <indusridhar@microsoft.com>

* Node os upgrade desc fix (#21636)

* Updating response abort abort apis

* Updating abort AP and MC examples

* Changes per CI checks

* Adding x-ms-long-running-operation-options

* Adding location header

* Fixing location url in abort examples

* Fixing headers and body for abort response

* Linter fix

* Lint fix

* Updating managedclusters.json responses

* Minor edit to node os upgrade description based on ARM review

Co-authored-by: InduSridhar <indusridhar@microsoft.com>

Co-authored-by: Paul Miller <pmiller@microsoft.com>
Co-authored-by: Thalia Wang <58485997+wenxuan0923@users.noreply.github.com>
Co-authored-by: Matthew Christopher <matthchr@users.noreply.github.com>
Co-authored-by: InduSridhar <InduSridhar@users.noreply.github.com>
Co-authored-by: InduSridhar <indusridhar@microsoft.com>

Alancere committed:
[Go] add go repo in readme.md (#21649)

ZhichengYang-MS committed:
[Synapse]add new apis of Pause/Resume for Synapse Link (#21240)

* add pause resume apis

* fix examples

* simplified changes

* fix version

* fix linDiff

* remove location info in url

houk-ms committed:
[Hub Generated] Publish private branch 'servicelinker/update-readme-apiversion' (#21635)

* add or modify files

* hide unused changes

* eliminate diff from private repo

dindurkhya committed:
ACSS Swagger Changes in PUT SVI API and child resources (#21604)

* @Shreyas : SAP Trans Changes for Swagger

* @Divyani: SAP Disk Customization - Adding DiskConfiguration in PUTSVI

* @Bharath: Adding LB, Storage details to child resources

ruowan committed:
Revert "ACSS Swagger Changes in PUT SVI API and child resources (#21604)" (#21662)

This reverts commit 58f076e3d73a63d60d71c5f57ba46acfdc40ecd0.

navba-MSFT committed:
{AzureDataProtection} fixes Azure/azure-rest-api-specs#21183 (#21551)

fixes Azure/azure-rest-api-specs#21183

The enum values for the storage settings are listed in the latest API as `ArchiveStore, SnapshotStore, VaultStore.`
However while providing `SnapshotStore` value it fail with `BMSUserErrorInvalidInput` error. Also while trying the same from PS cmdlet using

> New-AzDataProtectionBackupVaultStorageSettingObject -DataStoreType SnapshotStore -Type LocallyRedundant

**Error:**
```
Error: "Unable to match the identifier name 
SnapshotStore to a valid enumerator name. Specify one of the following enumerator names and try again:
ArchiveStore, OperationalStore, VaultStore"
```

This PR fixes the enum value of storage setting

navba-MSFT committed:
{AzureRecoveryServices} fixes Azure/azure-rest-api-specs#20953 (#21290)

* {AzureRecoveryServices} fixes Azure/azure-rest-api-specs#20953

fixes Azure/azure-rest-api-specs#20953

swagger link: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-12-01/bms.json#L4362

example link: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-12-01/examples/ResourceGuardProxyCRUD/PutResourceGuardProxy.json

In the swagger file, there is no parameter named resourceGuardResourceId in ResourceGuardProxy_Put while in the example there is. the parameter is needed to create.

* Update bms.json

Wzb123456789 committed:
[Network] Update readme.python.md (#21681)

* Update readme.python.md

* Update readme.python.md

damodaravadhani committed:
[EventHubs]: Adding missing descriptions (#21685)

* Adding missing descriptions

* fix

* Doc Fix

* Fixes

msyache committed:
Face PersonDirectory argument type mismatch fix (#21572)

* Fix Face PersonDirectory argument type mismatch

Update the Person related operations in PersonDirectory to consume
personId as GUID as opposed to string.
The change adheres to the existing pattern for Person Group and
Large Person Group, clarifies intended usage, and should reduce
confusion from the current discrepancy between returned GUIDs in
responses while the corresponding operation arguments are string.

* Fix person request type

* Fix AddPersonFace missing url in json

ashoknailwal committed:
remove required list from reverseshippingdetails since it has optional properties (#21625)

Wzb123456789 committed:
Update readme.python.md (#21706)

vineetmimrot committed:
[Hub Generated] Review request for Microsoft.Network to update version preview/2021-02-01-preview for NSP Link feature (#21617)

* Update NSP Link Swagger

* fix model validation

aronip committed:
Merging new API version from private pr repo https://github.com/Azure/azure-rest-api-specs-pr/blob/RPSaaSMaster/specification/connectedvmware/resource-manager/Microsoft.ConnectedVMwarevSphere/preview/2022-07-15-preview (#21658)

* Add 2022-07-15-preview API version from private repo.

* sync autorest files

* remove publicKey from VMProperties.

bexxx committed:
[Cognitive - AzureOpenAI] Review request for GA version 2022-12-01 (#21312)

* Add old files as new version to simplify diff

* Add suffix support

* Remove token based completion from inference

* Add embeddings response, add usage to responses

* Update GA API

* Add error property to entities

* Inline hyper parameters on fine tune creation

* Add custom word

* Address feedback from api review meeting

* Less readonly and enums modelled as string

* Add custom word

* Making swagger more explicit

* Use correct body parameter name, make error required in error response.

* Use correct format for urls.

* Try suppressing linter warnings

* Mior doc improvement for referenced identities

yareyes committed:
[Hub Generated + S360] Review request for Microsoft.SqlVirtualMachine to add version preview/2022-07-01-preview (#21564)

* Adds base for updating Microsoft.SqlVirtualMachine from version preview/2022-02-01-preview to version 2022-07-01-preview

* Updates readme

* Updates API version in new specs and examples

* Updating swagger to latest api

* Add default values for optional parameters

* Adding default to optional parameters

* Fixing ModelValidator

* Adding long running options

* Update response to async

* Removing headers

* Adding headers for deletion

* Updating location for Post

* Added prettier check

* Reverted changes from error codes

* Removing the location from POST

malpaniHMC committed:
[Hub Generated] Publish private branch 'shmalpani/traffic-controller-init' (#21674)

* traffic controller initial commit for PR

* updating frontend and association paths

* adding association definitions and fixing typos

* renaming directories

* updating readme with latest directory info

* updating directories to match tags

* adding get example

* adding example properties

* reorganizing properties replied with

* ensuring example matches with json definition

* adding frontend get example

* association get example

* adding GET examples

* adding PUT examples

* PUT examples

* frontend PUT and TrafficControllers List

* DELETE examples

* UPDATE examples

* spec version change + prettier

* changing API version

* linter errors

* x-ms-client-flatten property to avoid linter warns

* operationslist example and updating versioning

* prettier

* adding cadl file and patch/update semantics

* cadl commit

* using ResourceInstanceOperations

* descriptions updates

* examples typos

* solving examples errors

* move cadl file

* moving cadl

* prettier

* package json

* examples moved for cadl

* prettier

* moving examples

* configuration endpoints as read-only

* updating put example

* prettier

* renaming files, and making revision changes

* readme, linter, and examples revisions

* removing private IP

* Fixing linter errors

* prettier-fix

* updating pattern regex for parameters

Co-authored-by: Shivam Malpani <shmalpani@microsoft.com>

xiaoxuqi-ms committed:
add containerservice (#21713)

yanjungao718 committed:
change type for folder path of data flow staging (#21688)

pednekarpraj committed:
Fix s360 in reservation and scheduled action APIs (#21677)

* Fix s360 in reservation and scheduled action APIs

* Fix lint errors

* Fix linter error in RI recommendation example

* Fix lint errors

* Review comment - changed suppression rule name

PTaladay committed:
[Hub Generated] Review request for Microsoft.HealthcareApis to add version stable/2022-06-01 (#20116)

* Removed maximum constraint for offerThroughput to keep aligned with previously approved change for previous versions. This version was published while the approval was pending so it didn't get updated at that time.

* Fixes for model validation errors related to LRO_Response_HEADER.

* Added incorrect change to resolve LRO model validation. Updated.

* Another LRO fix.

* Fix formatting issue.

* Removing Resource header tag.

roslynlu committed:
[Hub Generated] Review request for Microsoft.StreamAnalytics to add version preview/2021-10-01-preview (#21632)

* Add blockchain to latest profile

* Add additional types

* update enum name

* update cluster max size and add default values

* add gateway message bus

* tenantId and principalId to readonly

* add blobpathprefix

* add file ref input

* add missed example files

* add 200 response code to scale

* add to 2020GA version

* change sizeWindow to int

* fix api version

Co-authored-by: Mark Cowlishaw <markcowl@microsoft.com>
Co-authored-by: Roslyn Lu <roslu@microsoft.com>

xfz11 committed:
add or modify files (#21687)

tongwu-sh committed:
Dev healthcareapis microsoft.healthcare apis 2022 10 01 preview (#19948)

* Adds base for updating Microsoft.HealthcareApis from version stable/2022-06-01 to version 2022-10-01-preview

* Updates readme

* Updates API version in new specs and examples

* add analytics connector

* add analytics connector example

* fix error

* fix error

* fix error

* fix error

* remove 202 for create

* fix bug

* add discriminators

* Add missing required type

* remove 2 properties

* remove additional properities

* add long running operations

* fix addition properties

* fix name pattern

* fix identity alert

* revert workspace name change

* fix location

* fix warning

* fix warning

* fix LRO error

* add async headers

* merge from stable version

Co-authored-by: Tong Wu (SH) <tongwu@microsoft.com>

catalinaperalta committed:
[AnomalyDetector] Add CADL specification for Anomaly Detector v1.1 (#21525)

* add univariate cadl definitions

* add cadl project yaml and main cadl

* add package.json

* add initial multivariate cadl definitions

* update cadl-project.yaml and package.json

* update route names

* add univariate azure core operation defs

* update multivariate models for correctness

* update routes to reflect service instead of using cadl azure operation signatures

* update multivariate models for correctness with original swagger

* a note about errors

* decorate request and response bodies with body decorator

* update package.json to point to next package versions

* fix routes to match swagger names

* add client.cadl

* remove client.cadl

* fix GetBatchDetectionResult def

* add namespace for univariate models

* Version AD (#21497)

* Update main.cadl

* fix versioning

* leave api version as required

Co-authored-by: Catalina Peralta <catalinaperaltah@hotmail.com>

* add extensible api version enum

* update cadl-project.yaml with python config

* update sdk version

* clean up routes

* clean up cadl-project.yaml

* clean up models

* update package.json to ref latest deps

* add newline

* add more models docs

* fix error response models

* rename get multivariate batch result

* Cadl dpg add convenience api (#21575)

* add @convenienceAPI

* move @convenienceAPI to client.cadl

* add placeholder examples file

* add generated openapi spec

* delete older swaggers

* update cadl-project.yaml

* update package.json

* Add cadl-csharp parameters for AnomalyDetector (#21647)

* Change cadl-project

* Add .net parameters

* Fix

* update error header uvad

* Make model-namespace false for .net (#21686)

* [AnomalyDetector] Archboard feedback (#21680)

* rename to train only

* archboard naming feedback

* rename response to result

* archboard renames mvad requests to options

* rename uvad request to options

* rename model

* rename model on operations

* more mvad renames

* update body naming

* update body param naming to not use a reserved keyword

* restructure cadl files

* move generated cadl

* update client.cadl

* update swagger output

* Update specification/cognitiveservices/AnomalyDetector/univariate/models.cadl

Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

* Update specification/cognitiveservices/AnomalyDetector/univariate/routes.cadl

Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

* Update specification/cognitiveservices/AnomalyDetector/multivariate/models.cadl

Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

* Update specification/cognitiveservices/AnomalyDetector/multivariate/routes.cadl

Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

* Update specification/cognitiveservices/AnomalyDetector/multivariate/models.cadl

Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

* Update specification/cognitiveservices/AnomalyDetector/multivariate/models.cadl

Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

* Update specification/cognitiveservices/AnomalyDetector/multivariate/models.cadl

Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

* Update specification/cognitiveservices/AnomalyDetector/multivariate/models.cadl

Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

* Update specification/cognitiveservices/AnomalyDetector/multivariate/models.cadl

Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

* make data schema enum extensible

* make errors on ModelInfo readonly

* update swagger

* update readme swagger refs

* add apiview to custom-words.txt

* update more refs in readme

* move existing examples up with CADL files

* configure swagger file name in emmiter

* format examples

* point to examples dir in cadl-project.yaml

* add generated swagger and examples

Co-authored-by: Laurent Mazuel <laurent.mazuel@gmail.com>
Co-authored-by: Weidong Xu <weidxu@microsoft.com>
Co-authored-by: Pan Shao <97225342+pshao25@users.noreply.github.com>
Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

fnuarnav committed:
[Hub Generated] Review request for Microsoft.ContainerInstance to add version preview/2022-10-01-preview (#21530)

* Adds base for updating Microsoft.ContainerInstance from version stable/2022-09-01 to version 2022-10-01-preview

* Updates readme

* Updates API version in new specs and examples

* added ConfidentialComputeProperties, and Confidential ContainerGroupSku

* added Untested confidential container example

* Add priority property to ContainerGroup

* Add examples with priority property

* removed IsolationType from ConfidentialComputeProperties

* fixed spellcheck issues

* Change priority description

Co-authored-by: Divya Bharadwaj Yellapragada <diyellap@microsoft.com>

ZianWang02 committed:
Add profile level check endpoint name availability API (#21646)

* Add profile level check endpoint name availability API

* Update afdx.json

ruowan committed:
Revert "Revert "ACSS Swagger Changes in PUT SVI API and child resources (#21604)" (#21662)" (#21663)

This reverts commit 495363bc011ce917f579adc1a5209073565d37f4.

minkefusiji committed:
Update doc (#21748)

* Update doc

* Remove suppress

* Remove suppress

* Remove suppress

* Add ad openapi description

* Update typo

Co-authored-by: jr-MS <81400625+jr-MS@users.noreply.github.com>

oravital7 committed:
[Hub Generated] Publish private branch 'oravital-dev-security-Microsoft.Security-2022-01-01-preview' (#21751)

* Update governance API docs (At scale)

* Fix model validation

* Fix lint

* Plain english style

* Minor

* Update content

* Renaming

* Comments

* Edit descriptions'

* Adding deletion operation result

* Prettier

* Remove LRO

minnieliu committed:
Update Media Composition Swagger to latest version (#21642)

* Update Media Composition Swagger

* Manually update the swagger to unblock

* Fix spec errors

* Prettify

* Prettify

* Updating swagger

* Prettify

Co-authored-by: Minnie Liu <peiliu@microsoft.com>

pansharm-ms committed:
[Purview - PDS] - 2021-01-01-preview (#20142)

* Add PDS swagger specification.

* Use default instead of multiple error codes.

* Add desctiptions.

* Fix error response model.

* Update PDS api specifications as per API guidelines

* Pretty.

* Correct example.

* fix dug property.

* Remove from custom words.

* Fix linting errors.

* Fix errors.

* Fix conflict.

* Fix examples.

* Fix linting errors.

* Linting errors.

* Linting errors.

* Fix operatio Id.

* Fix x-ms-pageable.

* Add filter description.

Co-authored-by: Gaurav Gupta <gauravgupta@microsoft.com>

chlowell committed:
Document that public_exponent applies only to MHSM (#21762)

andtu committed:
AAD Domain Services: Add 2022-12-01 API version (#21567)

* AAD Domain Services: Add 2022-12-01 API version

* Update all api-version in files to latest 2022-12-01 version

* Add SyncApplicationId property to domainservices.json

* pr review feedback on the property description

yuwzho committed:
[Hub Generated] Publish private branch 'appplatform/release/s213' (#21746)

* [AutoSync] 3a0ff3ca9 Merged PR 7162965: fix: add missing change

* Update readme.md

Co-authored-by: swagger-automation <swagger@microsoft.com>

shesamian committed:
[Hub Generated] Review request for Microsoft.RecoveryServices to add version stable/2022-10-01 (#21450)

* Adds base for updating Microsoft.RecoveryServices from version preview/2022-09-01-preview to version 2022-10-01

* Updates readme

* Updates API version in new specs and examples

* moving latest preview to stable

Co-authored-by: Himanshu Agarwal <himanshu.123j@gmail.com>

brrusino committed:
added duration format to support ISO8601 strings (#21711)

bdefoy committed:
Add more parameters to common types (#21420)

* Add more parameters to common types

* Put back the line at the end of the file

* Skip URL encoding on already-encoded parameter

agrbhuvnesh committed:
[Hub Generated] Publish private branch 'user/pri/2022-12-01' (#21717)

* add or modify files

* Fixed spellcheck validation err

* Fixed avocado validation err

* Fixed avocado validation err

* Fixed avocado validation err

* Fixed avocado validation err

* Fixed avocado validation err

* Fixed avocado validation err

* Fixed avocado validation err

* Fixed avocado validation err

* avacado fix

* avacado fix

* avacado fix

* avacado fix'

* avacado fix

* avacado fix

* revert custom-words.txt

* update custom-words.txt

* updated PreCheckResult

* removed SystemData

* removed SystemData

Co-authored-by: Aditya Ravishankar <ravishankara@microsoft.com>

shams11 committed:
[Hub Generated] Publish private branch 'RPSaaSMaster' (#21518)

* add or modify files

* Update FileSystems_CreateOrUpdate_MaximumSet_Gen.json

* Adding 2022-06-27 preview folder

* Adding custom-words.txt

* Adding Qumulo.Storage in ./custom.txt file

* Adding id in the response

* Fixed ModelValidation errors

* Fixed prettier issue

* Reverting FileSystems_CreateOrUpdate_MaximumSet_Gen.json

* Fixed prettier issue

* Adding cadl file

* Adding examples for cadl

* Removing readme file

* Adding read me files

* /Removing read me files

* Adding Read me all supported languages

* Changed python readme

* Removing npmrc and cadl-project.yaml

* Moved the cadl under resource-manager and added cadl to support both API versions

* Adding readme files

* Removing read me

* Adding .gitignore file

* Removing examples

* Renaming to cadl folder

* Renaming cadl folder to Qumulo.Management

* Formatted cadl file

* Formatted cadl file

* Adding examples in Qumulo.Management

* adding cadl-project.yaml

* Adding new package.json

* Reverting examples in Qumulo.Management

* Adding examples in Qumulo.Management

Co-authored-by: shamsthabrez <shamsthabrez@microsoft.com>

abhayohri committed:
[Hub Generated] Review request for Microsoft.ResourceHealth to add version stable/2022-05-01 (#21588)

* Adds base for updating Microsoft.ResourceHealth from version preview/2022-05-01-preview to version 2022-05-01

* Updates readme

* Updates API version in new specs and examples

* Changes to example files for ImpactedResources.

* was told to add a pattern restiction

* was told to add a pattern restiction

Co-authored-by: ABHISHEK SINGHAL <abhsinghal@microsoft.com>

MikhailTryakhov committed:
Adds base for updating Microsoft.Network from version stable/2022-07-01 to version 2022-09-01

MikhailTryakhov committed:
Updates readme

MikhailTryakhov committed:
Updates API version in new specs and examples

krishmi committed:
Added flowlog property in virtual network (#21790)

Co-authored-by: Krishna Mishra <krmishr@microsoft.com>

Khushboo-Baheti committed:
commit1 (#22111)

Co-authored-by: Khushboo Baheti <khbaheti@microsoft.com>

utbarn-ms committed:
adding auth status property to circuit (#22024)

nikhilpadhye1 committed:
resolving merge conflicts

nikhilpadhye1 committed:
Merge branch 'release-Microsoft.Network-2022-09-01' into nipadhye/firewallPacketCaptureFeature

nikhilpadhye1 committed:
Revert "resolving merge conflicts"

This reverts commit fbb282246a0fcc1d3ee89345eeffe19552ea60bd, reversing
changes made to 11a0fd1b157fe92fb0627c5bb1678f618b7742ce.

nikhilpadhye1 committed:
git status

nikhilpadhye1 committed:
Moving PR to 2022-09-01 api

nikhilpadhye1 committed:
Adding packet capture API from 2022-07-01 to 2022-09-01

nikhilpadhye1 committed:
fixing removed space

raosuhas commented on specification/network/resource-manager/Microsoft.Network/stable/2022-09-01/azureFirewall.json:
```diff
@@ -1148,6 +1210,127 @@
+        },
+        "sasUrl": {
+          "type": "string",
+          "description": "Upload capture location"
```
This should have x-ms-secret if it contains a sas url with secrets. [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret)

nikhilpadhye1 replied:
The SASURL does not contain secrets it will have only write create and add permissions to store packet capture on Cx storage account/container.

raosuhas commented on specification/network/resource-manager/Microsoft.Network/stable/2022-09-01/azureFirewall.json:
```diff
@@ -437,7 +437,69 @@
+            "description": "Accepted and the operation will complete asynchronously."
+          },
+          "200": {
+            "description": "The packet capture request was successfully triggered."
```
>The packet capture request was successfully triggered [](http://example.com/codeflow?start=28&length=53)

nit : update description  , 200 should be for the final successful state not when it is trigerred

nikhilpadhye1 committed:
addressing pr comments

nikhilpadhye1 commented:
@raosuhas  can you please take a look at the response to comments and fix of NIT and approve or advice on further changes?

utbarn-ms committed:
Make auth status readonly (#22365)

* make auth status read only

* fixing model validation

* prettier fix

nikhilpadhye1 committed:
errors with parenthesis

nikhilpadhye1 committed:
fixing definition

nikhilpadhye1 committed:
fix

tejasshah7 committed:
Add support for State flag in Custom Rule (#22457)

prateek2211 committed:
Fix LRO header model validation (#22506)

nikhilpadhye1 committed:
fixing location error

nikhilpadhye1 committed:
fixing azureFirewallName error

nikhilpadhye1 committed:
adding string return for packetcapture call

prateek2211 committed:
Add new status code for application gateway custom error page (#22151)

* Add new status code for application gateway custom error page

* Fix prettier

zizw123 commented:
Please check the lint error in https://github.com/Azure/azure-rest-api-specs/pull/21080/checks?check_run_id=11211115269

nikhilpadhye1 commented:
> Please check the lint error in https://github.com/Azure/azure-rest-api-specs/pull/21080/checks?check_run_id=11211115269

Adding the pattern linter error is existing for firewall. Adding the pattern may cause the feature to not be accessible to existing azure firewall customers. If you have a suggestion on how to safely circumvent this possibility, please let me know. Need a solution as this is a blocker for the PR. 
@zizw123

prateek2211 committed:
Fix missing resource id in application gateway list example (#22509)

nikhilpadhye1 committed:
Merge remote-tracking branch 'upstream/release-Microsoft.Network-2022-09-01' into nipadhye/firewallPacketCaptureFeature

lirenhe commented:
@nikhilpadhye1, I already see the PR to merge NRP 2022-09-01 to main branch. Please work with @MikhailTryakhov if this PR is needed for 2022-09-01, otherwise, you need to wait for the next release.

nikhilpadhye1 committed:
Adding pattern to fix lintdiff and modelvalidation error

nikhilpadhye1 committed:
Adding pattern to fix lintdiff and modelvalidation error

nikhilpadhye1 committed:
adding response for packet capture request

nikhilpadhye1 committed:
prettier check

rkmanda commented on specification/network/resource-manager/Microsoft.Network/stable/2022-09-01/azureFirewall.json:
```diff
@@ -437,6 +437,80 @@
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/azureFirewalls/{azureFirewallName}/packetCapture": {
```
>packetCapture [](http://example.com/codeflow?start=135&length=13)

make it a verb capturePacket?

nikhilpadhye1 replied:
Want to keep consistency between features and repos.

nikhilpadhye1 committed:
removing response for this version and will bring it in upcoming api versions

nikhilpadhye1 committed:
removing 200 response

nikhilpadhye1 committed:
Merge branch 'release-network-Microsoft.Network-2022-11-01' into nipadhye/firewallPacketCaptureFeature

nikhilpadhye1 committed:
changing return type

raych1 commented:
/azp run

raych1 commented:
@nikhilpadhye1, can you follow instructions to get breaking change approval?

nikhilpadhye1 commented:
This PR was signed off by arm team for 2022-09-01 but approvals didnt make the cut date. Closing this pr and making new one for 2022-11-01.
